spec/rubocop/cop/layout/line_length_spec.rb

```diff
@@ -261,6 +261,20 @@ def test_some_other_long_test_description_which_exceeds_length
       RUBY
     end
 
+    context 'and SplitStrings option is enabled' do
+      let(:cop_config) do
+        { 'Max' => 80, 'AllowHeredoc' => true, 'SplitStrings' => true }
+      end
+
+      it 'does not register an offense' do
+        expect_no_offenses(<<~'RUBY')
+          <<~MESSAGE
+            #{'hello' * 1} #{'world' * 2} #{'hello' * 1} #{'world' * 2} #{'hello' * 1} #{'world' * 2}
+          MESSAGE
+        RUBY
+      end
```

Comment:
There should be a test for `AllowHeredoc: false` as well, that expects the line is too long and there is no correction. You can add it to the context starting on line 532 (following the other heredoc context on line 760 would be a good spot).

Assuming `Max` is set to 40 (as per the aforementioned context):

```ruby
it 'registers an offense and corrects' do
  expect_offense(<<~'RUBY')
    <<~MESSAGE
      #{'hello' * 1} #{'world' * 2} #{'hello' * 1}
                                            ^^^^^^ Line is too long. [46/40]
    MESSAGE
  RUBY

  expect_no_corrections
end
```